src/utils/netutils.cpp

```diff
@@ -811,3 +965,69 @@ bool get_local_ip_address(IpAddress& client_address, int fd) noexcept
 
     return true;
 }
+
+bool find_probe_client(std::string_view probe_client_addresses,
+                       zstring_view source_ip,
+                       bool is_ipv6)
+{
+    if (probe_client_addresses.empty())
+    {
+        return false;
+    }
+
+    sockaddr_storage source_ss;
+
+    if (!get_in_addr_from_ip(source_ss, source_ip.c_str(), is_ipv6))
+    {
+        return false;
+    }
+
+    char probe_ip_buf[INET6_ADDRSTRLEN] = {};
+    auto try_fill_buffer_and_compare_ip = [&](auto start, auto end)
+    {
+        /* check substring size is greater than 0 and less
+           than IPv6 max size (i.e a valid IP in theory) */
+        if (end <= start || end - start >= sizeof(probe_ip_buf))
+        {
+            return false;
+        }
+
+        auto i = 0;
+
+        while (start < end)
+        {
+            probe_ip_buf[i++] = probe_client_addresses[start++];
+        }
+
+        probe_ip_buf[i] = '\0';
+
+        if (compare_binary_ip(source_ss, probe_ip_buf, is_ipv6))
+        {
+            return true;
+        }
+
+        return false;
+    };
+    auto pos = 0;
+    auto found = 0;
+
+    while ((found = probe_client_addresses.find(',', pos))
+           != probe_client_addresses.npos)
```

Comment:
```cpp
for (chars_view addr : split_with(probe_client_addresses, ','))
```

Regarding auto for `pos` and `found`, that's a bad idea. `std::string::find()` takes and returns a `std::string::size_type` (=`std::size_t`) while found and pos are of type `int`.